Gui/python/ArduinoWidget.py

```diff
@@ -71,114 +73,80 @@ def createArduino(self):
-            print(self.ResourcesList)
-            self.getDeviceName()
-            return list(self.deviceMap.keys())
-        except Exception as err:
-            logger.error("Failed to list all resources: {}".format(err))
-            self.ResourcesList = ()
-            return self.ResourcesList
 
-    def getDeviceName(self):
-        self.deviceMap = {}
-        for device in self.ResourcesList:
-            try:
-                pipe = subprocess.Popen(
-                    [
-                        "udevadm",
-                        "info",
-                        " --query",
-                        "all",
-                        "--name",
-                        device.lstrip("ASRL").rstrip("::INSTR"),
-                        "--attribute-walk",
-                    ],
-                    stdout=subprocess.PIPE,
-                    stderr=subprocess.STDOUT,
-                )
-                raw_output = pipe.communicate()[0]
-                vendor_list = [
-                    info
-                    for info in raw_output.splitlines()
-                    if b"ATTRS{idVendor}" in info or b"ATTRS{vendor}" in info
-                ]
-                product_list = [
-                    info
-                    for info in raw_output.splitlines()
-                    if b"ATTRS{idProduct}" in info or b"ATTRS{product}" in info
-                ]
-                idvendor = (
-                    vendor_list[0]
-                    .decode("UTF-8")
-                    .split("==")[1]
-                    .lstrip('"')
-                    .rstrip('"')
-                    .replace("0x", "")
-                )
-                idproduct = (
-                    product_list[0]
-                    .decode("UTF-8")
-                    .split("==")[1]
-                    .lstrip('"')
-                    .rstrip('"')
-                    .replace("0x", "")
-                )
-                deviceId = "{}:{}".format(idvendor, idproduct)
-                pipeUSB = subprocess.Popen(
-                    ["lsusb", "-d", deviceId],
-                    stdout=subprocess.PIPE,
-                    stderr=subprocess.STDOUT,
-                )
-                usbInfo = pipeUSB.communicate()[0]
-                deviceName = usbInfo.decode("UTF-8").split(deviceId)[-1].lstrip(" ")
 
-                if deviceName == None:
-                    logger.warning("No device name found for {}:".format(device))
-                    # self.deviceMap[device] = device
-                elif "Arduino" in deviceName:
-                    self.deviceMap[deviceName] = device
-                else:
-                    pass
-            except Exception as err:
-                logger.error("Error found:{}".format(err))
-                # self.deviceMap[device] = device
+            compileResult=subprocess.run(
+            "$GUI_dir/bin/arduino-cli compile --fqbn arduino:avr:uno $GUI_dir/FirmwareImages/relay_box_firmware/",
```

Comment:
Why is there the exact same code in 2 places?

Reply:
The "upload script" at Arduinowidget is for uploading the temperature monitoring script. And the one at QtApplication is for uploading the relay box script. The reason why I can't use "uploade script" in QtAplication for uploading temperature monitor script is that the button "Upload script" inside Arduinowidget can only connect the self function inside class ArduinoWidget.